### PR TITLE
[Alter] Add more schema change to varchar type

### PR DESCRIPTION
**Pull request by WingsGo**
#2776

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]
+1. LGTM

File: be/test/olap/schema_change_test.cpp [hunk 1]
```diff
@@ -212,6 +212,47 @@ class TestColumn : public testing::Test {
         ASSERT_EQ(_column_writer->create_row_index_entry(), OLAP_SUCCESS);
     }
 
+    void test_convert_to_varchar(std::string type_name, int type_size, const std::string& value, OLAPStatus expected_st) {
```
Comment (chaoyli) [inline be/test/olap/schema_change_test.cpp]: std::string& type_name

**Review by chaoyli — CHANGES_REQUESTED** [review]

**Comment by morningman** [comment]
Plz update the help doc of alter table. Add new conversions

**Comment by WingsGo** [comment]
@imay  @morningman  @chaoyli  Plz review this PR~

**Review by chaoyli — APPROVED** [review]

**Pull request merged**